anonip.py

```diff
@@ -80,22 +80,43 @@ def __init__(
         """
         Main class for anonip.
 
-        :param columns: list of ints
+        :param columns: list of int, 1-based column numbers
         :param ipv4mask: int
         :param ipv6mask: int
         :param increment: int
         :param delimiter: str
         :param replace: str
         :param skip_private: bool
         """
-        self.columns = columns if columns else [1]
+        columns = columns if columns else [1]
+        # change columns to be 0-based
+        self.columns = [c - 1 for c in columns]
+        self._prefixes = [None] * 7  # 1-based
         self.ipv4mask = ipv4mask
         self.ipv6mask = ipv6mask
         self.increment = increment
         self.delimiter = delimiter
         self.replace = replace
         self.skip_private = skip_private
 
+    @property
+    def ipv4mask(self):
+        return self._ipv4mask
+
+    @ipv4mask.setter
```

Comment:
What is the reason for using getters and setters here?

Couldn't we just have this in `__init__()`?:

```python
self.prefixes = {
    4: 32 - ipv4mask,
    6: 128 - ipv6mask
}
```

Reply:
Nevermind, I get it. Still I'd prefer having a dict `self.prefixes` instead of the list.

Reply:
My intention was to gain speed, as I assume accessing elements using an index into an array is faster then fetching a hash first. But as https://wiki.python.org/moin/TimeComplexity as well as my own tests show: dicts are as fast as lists in our case (and maybe even a small tick faster).

(Thinking in C-code I can not understand this, but, well, I did not trace the CPython code in detail ;-)

So I will change the code to use dicts.

Reply:
For the records, her is the code I used for testing
```
import timeit

ipv4mask = 11
ipv6mask = 83
prefixes_dict = {
    4: 32 - ipv4mask,
    6: 128 - ipv6mask
}
prefixes_list = [None, None, None, None,
                 32 - ipv4mask, None, 128 - ipv6mask]

def from_dict(): return prefixes_dict[4]

def from_list(): return prefixes_list[4]

timeit.timeit('from_dict()', globals=globals())
timeit.timeit('from_list()', globals=globals())


sum(timeit.timeit('from_dict()', globals=globals()) for i in range(1000))
sum(timeit.timeit('from_list()', globals=globals()) for i in range(1000))
```

Reply:
Yeah, dicts are hashtables in python and therefore very performant.